src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ChunkOutputFrame.java

```diff
@@ -25,13 +26,62 @@ void loadUrl(String url, Command onCompleted)
       onCompleted_ = onCompleted;
       super.setUrl(url);
    }
+   
+   /**
+    * Loads a URL after a fixed amount of time.
+    * 
+    * @param url The URL to load.
+    * @param delayMs The number of milliseconds to delay before loading the URL.
+    * @param onCompleted The command to run after the time has passed *and* the
+    *   URL had been loaded.
+    */
+   void loadUrlDelayed(String url, int delayMs, 
+         Command onCompleted)
+   {
+      // prevent stacking timers
+      if (timer_ != null && timer_.isRunning())
+         timer_.cancel();
+      
+      onCompleted_ = onCompleted;
+      url_ = url;
+
+      timer_ = new FrameLoadTimer();
```

Comment:
It's somewhat academic since there these objects don't get re-used (i.e. `loadUrl` is only ever called once), but I agree that's a better design! Updated in https://github.com/rstudio/rstudio/commit/8054e585dbb8eaa13eb3a6392828db83156552bc.